### PR TITLE
For empty list/tuple, use floatX for the dtype instead of the numpy d…

### PR DESCRIPTION
**Pull request by nouiz**
…efault (float64)

### CODE REVIEW

**Comment by nouiz** [comment]
Should I bump dev number to dev4?

File: theano/tensor/tests/test_basic.py [hunk 1]
```diff
@@ -7118,6 +7118,16 @@ def test_memmap(self):
         new_inp[...] = inp
         x = as_tensor_variable(new_inp)
 
+    def test_empty_dtype(self):
+        old = theano.config.floatX
+        for dtype in ['float16', 'float32', 'float64']:
+            try:
```
Comment (abergeron) [inline theano/tensor/tests/test_basic.py]: What about change_flags?

**Comment by abergeron** [comment]
Globally, I am ok with this.

File: theano/tensor/tests/test_basic.py (hunk 1 above)
Comment (nouiz) [inline theano/tensor/tests/test_basic.py]: It is only a decorator in Theano right now. I don't want to wait for the RNN PR to be able to use the context manager. My first version was with change_flags as a context manager as I taught it was merged.

**Pull request merged**